docs/content/concepts/ops-jobs-graphs/jobs.mdx

```diff
@@ -190,19 +201,27 @@ Refer to the [Partitions documentation](/concepts/partitions-schedules-sensors/p
 You can supply a <PyObject object="ConfigMapping" />. This allows you to expose a narrower config interface to your job. Instead of needing to configure every op and resource individually when launching the job, you can supply a smaller number of values to the outer config, and the <PyObject object="ConfigMapping" /> can translate it into config for all the job's ops and resources.
 
 ```python file=/concepts/ops_jobs_graphs/jobs_with_config_mapping.py
-from dagster import config_mapping, job, op
+from dagster import Config, RunConfig, config_mapping, job, op
+
 
+class DoSomethingConfig(Config):
+    config_param: str
 
-@op(config_schema={"config_param": str})
-def do_something(context):
-    context.log.info("config_param: " + context.op_config["config_param"])
+
+@op
+def do_something(context, config: DoSomethingConfig) -> None:
+    context.log.info("config_param: " + config.config_param)
 
 
-@config_mapping(config_schema={"simplified_param": str})
-def simplified_config(val):
-    return {
-        "ops": {"do_something": {"config": {"config_param": val["simplified_param"]}}}
-    }
+class SimplifiedConfig(Config):
+    simplified_param: str
+
+
+@config_mapping
+def simplified_config(val: SimplifiedConfig) -> RunConfig:
+    return RunConfig(
+        ops={"do_something": DoSomethingConfig(config_param=val.simplified_param)}
+    )
```

Comment:
Do we want to keep the old examples around as well? cc: @erinkcochran87

Reply:
A downside to keeping the old examples around is that we need to continue to maintain the code and tests. If we think there's value in that, we could put these examples into tabs so folks can toggle between the new and the old, ex:

```
| Pydantic config | Legacy config |
```

We could remove the tabs and old examples in a future version, so we're not totally taking things away or making them difficult to find.

Alternatively, we didn't keep many (if any?) examples of `@repository` around when we did the `Definitions` migration, except for the dedicated Repository page. If we want to be consistent in our doc updates - and for this to be The Way users do this thing - it may be best to remove them.

Thoughts @schrockn? I realize this isn't a direct answer to your question, but there's some product strategy to consider here outside of just doc updates.

Reply:
Understood! Then lets go with having new examples only.

FWIW, I think it'd be good to validate a theory like that at some point. We can probably do some work in Google Analytics at some point for custom events, or even implement event capture like Snowplow. Definitely not a near-term need, but it'd be great to have more interaction data like this to validate design choices.